File: 451-sort-characters-by-frequency/sort-characters-by-frequency.java
class Solution {
    public String frequencySort(String s) {
        HashMap<Character,Integer> mp=new HashMap<>();
        for(char ch:s.toCharArray()){
            mp.put(ch,mp.getOrDefault(ch,0)+1);
        }
        Character[] c=new Character[s.length()];
        for(int i=0;i<s.length();i++){
            c[i]=s.charAt(i);
        }
        Arrays.sort(c,(a,b)->{
            int fa=mp.get(a),fb=mp.get(b);
            if(fb!=fa){
                return fb-fa;
            }
            return a-b;
        });
        StringBuilder sb=new StringBuilder();
        for(char ch:c){
            sb.append(ch);
        }
        return sb.toString();
    }
}

File: 451-sort-characters-by-frequency/sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        map<char,int> mp;
        for(char ch:s){
            mp[ch]++;
        }
        sort(s.begin(),s.end(),[&](int a,int b){
            if(mp[a]!=mp[b]){
                return mp[a]>mp[b];
            }
            return a<b;
        });
        return s;
    }
};